### PR TITLE
add particle adaptor options in SENSEI tutorials

### PR DESCRIPTION
**Pull request by c-wetterer-nelson**
There are a few CMake infrastructure upgrades made here to enable using the new SENSEI particle data adaptors in the SENSEI/Advection_AmrLevel tutorial. 
1. Add option to build against a local AMReX library rather than downloading a pinned Github repo and building it.
2. Move the SENSEI tutorial CMake files to the newer tutorial api used elsewhere

Next, the key addition here adds example particle container data adapting to the SENSEI/Advection_AmrLevel tutorial. This includes some new SENSEI scripts demonstrating using the new Particle Data Adaptors from https://github.com/AMReX-Codes/amrex/pull/2016

### CODE REVIEW

**Comment by atmyers** [comment]
Note, needs https://github.com/AMReX-Codes/amrex/pull/2016 to be merged first.

**Comment by burlen** [comment]
@c-wetterer-nelson I've got some patches for this. could you give me permission to push to your fork?

**Comment by c-wetterer-nelson** [comment]
@burlen and @atmyers, I think this is ready for review! let me know what you think

**Comment by ax3l** [comment]
Hi @c-wetterer-nelson, you might need to update the CI for HIP as we did in AMReX: https://github.com/AMReX-Codes/amrex/pull/2031

Background: `hipcc` with RDC flags (which you probably need for SENSEI?) is very flaky. Luckily the future is about using AMD's or upstream `clang++ -x hip` directly, as in the linked PR. Yet there is a nifty issue with `clang++ -x hip`, AMD's CMake scripts and Fortran, which will only be fixed in ROCm 4.4+: https://github.com/ROCm-Developer-Tools/HIP/pull/2280

So you current options are for HIP compilation with CMake:
- `hipcc`: but without RDC (not an option for SENSEI I guess?)
- `clang++` from AMD: supports RDC, but without Fortran until ROCm 4.4 is out
Using clang/clang++ directly is the future, the Fortran issue is temporary until they release https://github.com/ROCm-Developer-Tools/HIP/pull/2280

TL;DR: just copy the `hip.yaml` from AMReX mainline to here :)

**Comment by atmyers** [comment]
You can also fix this by pulling and merging with main.

**Comment by burlen** [comment]
@c-wetterer-nelson Once you've applied changes from main (ie. fix the hip issue) and you've put it through its paces re makefile and cmake build, I think it's all set. Thanks!

**Comment by c-wetterer-nelson** [comment]
Merged with the latest development on main. Everything seems to be working as expected on my machine!

**Review by atmyers — APPROVED** [review]

**Comment by atmyers** [comment]
Thank you for this PR!
